lib/grandpa/message_handler.go

```diff
@@ -53,11 +53,22 @@ func (h *MessageHandler) handleMessage(from peer.ID, m GrandpaMessage) (network.
 	case *CommitMessage:
 		return nil, h.handleCommitMessage(msg)
 	case *NeighbourMessage:
+		// It seems like we can afford to not retry handling neighbour message
+		// if it errors.
 		return nil, h.handleNeighbourMessage(msg)
 	case *CatchUpRequest:
+		// CatchUpRequest seems like something that can be dropped, if we fail
+		// to process it
 		return h.handleCatchUpRequest(msg)
 	case *CatchUpResponse:
-		return nil, h.handleCatchUpResponse(msg)
+		err := h.handleCatchUpResponse(msg)
+		// TODO: Retry for which errors
+		if err != nil {
+			// TODO: If I can directly access tracker, why are we using `in` channel for
+			// networkVoteMessage
+			h.grandpa.tracker.addCatchUpResponse(msg)
+		}
```

Comment:
I think we only want to add the catch up response to the tracker for certain errors, such as if a block is not yet synced, eg the blocktree `ErrNodeNotFound`